Data/Primitive/Array.hs

```diff
@@ -586,18 +586,28 @@ mapArray' f a =
 
 -- | Create an array from a list of a known length. If the length
 -- of the list does not match the given length, this throws an exception.
+
+-- Note [fromListN]
+-- ~~~~~~~~~~~~~~~~
+-- We want arrayFromListN to be a "good consumer" in list fusion, so we define
+-- the function using foldr and inline it to help fire fusion rules.
+-- If fusion occurs with a "good producer", it may reduce to a fold on some
+-- structure. In certain cases (such as for Data.Set) GHC is not be able to
```

Comment:
I think it is best to move this sentence ("`In certain cases (such as for Data.Set) GHC is not be able to ...`") to a comment into the body of the function, since that is not relevant information for the user.
```suggestion
-- structure. In certain cases (such as for Data.Set) GHC is not able to
```

Reply:
Which "user" do you mean? This comment block is not part of the Haddocks.

Reply:
Oh yeah, you are right. I did not notice an empty new line on line number 589 between the haddock and the comment.